crates/neon-sys/native/src/neon.cc

```diff
@@ -522,17 +522,23 @@ extern "C" void Neon_Task_Schedule(void *task, Neon_TaskPerformCallback perform,
   neon::queue_task(internal_task);
 }
 
-extern "C" void* Neon_ThreadSafeCallback_New(v8::Local<v8::Value> self, v8::Local<v8::Function> callback) {
+extern "C" void* Neon_EventHandler_New(v8::Local<v8::Function> callback) {
```

Comment:
This was a nice idea, but unfortunately it's not future-proof for N-API, which doesn't have an equivalent of `isolate::GetCurrent()` and `isolate->GetCurrentContext()`. And in fact, this means the `bind` API is also not future-proof for N-API.

What this means is, we really need the `EventHandler` to take a Neon context as an explicit argument to its constructor(s).

So the best path forward is going to be: let's just drop this down to a single constructor, which takes three arguments: `cx`, `self`, and `callback`. It's a noticeably less convenient API, but I think that's OK to get started. Down the road, we can experiment with convenience methods of `cx` (just like the other convenience constructors such as `cx.string(...)` etc), like `cx.event_handler(...)`.

But let's wait on the convenience forms and for now just have one method in the API:

```rust
EventHandler::new(cx, self, callback)
```

Reply:
@dherman Should be good now, please take a final look! 😄